packages/server/src/threads/automation.ts

```diff
@@ -192,6 +203,57 @@ class Orchestrator {
     }
   }
 
+  handleLooping(
+    loopSteps: LoopStep[] | undefined,
+    iterationCount: number,
+    loopStep: LoopStep | undefined,
+    loopStepNumber: number,
+    index: number,
+    originalStepInput: any,
+    step: any
+  ): void {
+    const tempOutput = initializeTempOutput(loopSteps, iterationCount)
+    const item = getLoopItems(loopStep?.inputs.binding, loopStep?.inputs.option)
+    updateContext(this._context, loopStepNumber, item, index)
+
+    if (
+      !typecastLoopInput(
+        loopStep,
+        loopStepNumber,
+        step,
+        tempOutput,
+        this.updateContextAndOutput.bind(this)
+      )
+    ) {
+      loopStep = undefined
+      loopSteps = undefined
+    }
+
+    originalStepInput = replaceFakeBindings(originalStepInput, loopStepNumber)
+
+    if (checkMaxIterations(index, loopStep?.inputs.iterations)) {
+      handleMaxIterations(
```

Comment:
Similar to `updateContext`, this is only used in one place. I'd just inline this back how it was personally, especially considering this function takes 3 `any`s and an unspecified `Function`.